### DISCUSSION TITLE
What option to pick for a full-stack app. (React, Rest API)

### DISCUSSION THREAD

**Discussion by jmunckhof** [ory/.github#30] [Q&A] [answered]
Hey guys,

I want to switch from Auth0 to Ory but I am not sure what solution to pick for my app.

I am building a NextJS front-end with a Java Spring Boot back-end rest api. Should I use Hydra? Or is Kratos enough and how to implement this in a backend system.

Thanks

**Comment by aeneasr** [comment, marked as answer]
Hey :) Welcome to Ory! Most likely it's enough to use Ory Identities (Ory Kratos). If you want a less painful migration from Auth0, you can also use Ory OAuth2 (Ory Hydra) and Ory Identities (Ory Kratos) together! This works out of the box when you run on the Ory Network (https://console.ory.sh). Here's a guide: https://www.ory.sh/run-oauth2-server-open-source-api-security/

**Reply by vinckr** [reply]
Hey @jmunckhof ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
